From 811d4f4c84ffd94e7fcd6f4677a5788e5df220b7 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Tue, 16 Sep 2014 20:18:57 -0400
Subject: [fs] nfsd: clean up reset_union_bmap_deny

Message-id: <1410898849-22126-54-git-send-email-bfields@redhat.com>
Patchwork-id: 95708
O-Subject: [Patch RHEL7 053/165] nfsd: clean up reset_union_bmap_deny
Bugzilla: 1044798 1123721 1141932
RH-Acked-by: Benjamin Coddington <bcodding@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Jeff Layton <jlayton@primarydata.com>

Fix the "deny" argument type, and start the loop at 1. The 0 iteration
is always a noop.

Signed-off-by: Jeff Layton <jlayton@primarydata.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 3b84240a7b756e3fea8eaea5a29e7c10afbd0a47
Bugzilla: 1044798
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 0a54fc9..5f72947 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -4266,10 +4266,11 @@ static inline void nfs4_stateid_downgrade(struct nfs4_ol_stateid *stp, u32 to_ac
 }
 
 static void
-reset_union_bmap_deny(unsigned long deny, struct nfs4_ol_stateid *stp)
+reset_union_bmap_deny(u32 deny, struct nfs4_ol_stateid *stp)
 {
  int i;
- for (i = 0; i < 4; i++) {
+
+ for (i = 1; i < 4; i++) {
   if ((i & deny) != i)
    clear_deny(i, stp);
  }
-- 
1.7.1